common/membership/hashring.go

```diff
@@ -265,10 +273,11 @@ func (r *ring) refreshRingWorker() {
 		select {
 		case <-r.shutdownCh:
 			return
-		case <-r.refreshChan: // local signal or signal from provider
+		case event := <-r.refreshChan: // local signal or signal from provider
 			if err := r.refresh(); err != nil {
 				r.logger.Error("refreshing ring", tag.Error(err))
 			}
+			r.notifySubscribers(event)
```

Comment:
As mentioned in the other comment, let's change refresh to return a bool to indicate refresh occurred.

```
refreshed, err := r.refresh()
if err != nil {
  r.logger.Error("failed to refresh ring", tag.Error(err))
}
if refreshed && err == nil {
  r.notifySubscribers(event)
}
```

Reply:
done